doc/developer/workflow.rst

```diff
@@ -471,6 +471,12 @@ checkpatch.sh
    When convenient it is preferred that these be cleaned up inline, but this is
    not required.
 
+   In general, a developer should heed the information reported by checkpatch.
+   However, some flexibility is needed for cases where human judgement yields
+   better clarity than the script. Accordingly, it may be appropriate to
+   ignore some checkpatch.sh warnings or errors per discussion among the
+   submitter(s) and reviewer(s) of a change.
```

Comment:
Perhaps add something like "Misreporting of Errors by the script is possible. When this occurs, a patch for checkpatch should be added to the pull request."